### PR TITLE
Fix invalid read of size 1 under parse_at_list():lib/Libcmds/parse_at.c

### PR DESCRIPTION
**Pull request by nithinj**
#### Bug/feature Description
* *Fix invalid read of size 1 under parse_at_list():lib/Libcmds/parse_at.c*

#### Cause / Analysis / Design
* *If input 'list' = ",", then l = strdup(list"), then c = l, then s = c = ',', and b gets "c - 1" which is before the first element of strdup-ed 'list'.*

#### Solution Description
* *Added a check to ensure that we are not accessing any element before the first entry in strdup-ed 'list'.*

#### Testing logs/output
Manual testing makes sense as we need to reproduce using valgrind.
[server_after.log](https://github.com/PBSPro/pbspro/files/2467533/server_after.log)
[server_before.log](https://github.com/PBSPro/pbspro/files/2467534/server_before.log)
[make.sh.log](https://github.com/PBSPro/pbspro/files/2467566/make.sh.log)
[parse_at_list.c.txt](https://github.com/PBSPro/pbspro/files/2467567/parse_at_list.c.txt)
[valgrind.sh.txt](https://github.com/PBSPro/pbspro/files/2467568/valgrind.sh.txt)

#### Checklist:

- [x] I have joined the **[pbspro community forum](http://community.pbspro.org/)**.
- [x] My pull request contains a **single, signed** commit. See **[setting up gpg signature](https://pbspro.atlassian.net/wiki/display/DG/Signing+Your+Git+Commits)**.
- [x] My code follows the **[coding style](https://pbspro.atlassian.net/wiki/display/DG/Coding+Standards)** of this project.
- [x] My change requires project documentation. See **[required documentation checklist](https://pbspro.atlassian.net/wiki/display/DG/Checklist+for+Developing+Features+and+Bug+Fixes)** for details.
   - [ ] I have added documentation in the **[project documentation area](https://pbspro.atlassian.net/wiki/display/PD)**.
- [ ] I have added new **PTL test(s) to my commit**. (See **[using PTL for testing](https://pbspro.atlassian.net/wiki/display/DG/Using+PTL+for+Testing)**) *(or)*
   - [x] I have added  **manual test(s) to this pull request and explained why PTL is not appropriate** for this case.
- [x] All new and existing automated tests have passed. (See **[running automated PTL tests](https://pbspro.atlassian.net/wiki/display/DG/PTL+Quick+Start+Guide)**).
- [x] I have attached **test logs to this pull request** as evidence of testing/verification.

__***For further information please visit the [Developer Guide Home](https://pbspro.atlassian.net/wiki/display/DG/Developer+Guide+Home).***__

CC: @bayucan

### CODE REVIEW

**Comment by sakshamgarg** [comment]
The fix looks good to me.

File: src/lib/Libcmds/parse_at.c [hunk 1]
```diff
@@ -182,7 +182,7 @@ parse_at_list(char *list, int use_count, int abs_path)
 			;
 
 		/* Drop any trailing blanks */
-		for (b = c - 1; isspace(*b); b--)
+		for (b = c - 1; b >= l && isspace(*b); b--)
```
Comment (mike0042) [inline src/lib/Libcmds/parse_at.c]: I would suggest the following:
```
for (b = c - 1; (b > 0) && isspace(*b); b--)
```

Comment (nithinj) [reply src/lib/Libcmds/parse_at.c]: This sanity check ensures that the address of `b` is never below the address of `l` which is the original string we are traversing.

Comment (mike0042) [reply src/lib/Libcmds/parse_at.c]: Sorry, that looked like a one. Please do add parenthesis around (b >= l)

Comment (bhroam) [reply src/lib/Libcmds/parse_at.c]: I'd consider using a different variable than a lowercase ell.  It just looks too much like a one.

File: src/lib/Libcmds/parse_at.c [hunk 2]
```diff
@@ -182,7 +182,7 @@ parse_at_list(char *list, int use_count, int abs_path)
 			;
 
 		/* Drop any trailing blanks */
-		for (b = c - 1; isspace(*b); b--)
+		for (b = c - 1; b >= list_dup && isspace(*b); b--)
```
Comment (mike0042) [inline src/lib/Libcmds/parse_at.c]: Please add parenthesis around (b >= list_dup) as per the style guide.

Comment (nithinj) [reply src/lib/Libcmds/parse_at.c]: I missed your previous comment. I've included parenthesis with my recent commit.

**Comment by nithinj** [comment]
Hi @mike0042 / @bhroam ,
I've rebased this branch. Please take this further.

**Comment by bhroam** [comment]
@nithinj, @mike0042 is your maintainer.  You should get another reviewer from your team.  I did not review your PR.  I only provided a comment.

**Comment by bhroam** [comment]
Your branch is out of date.  Please rebase again.

**Comment by nithinj** [comment]
I've rebased it. Please check.

**Pull request merged**